feat: Update OpenAI spec to include image url in message content (#113)
Lightning-AI/LitServe: README.md

    server = ls.LitServer(GPT2LitAPI(), spec=ls.OpenAISpec())
    server.run(port=8000)
```
&nbsp;

> LitServe's OpenAISpec also enables capability to handle images in the input. Below is an example of how to set this up using LitServe.

Reviewer: ```suggestion
LitServe's `OpenAISpec` can also handle images in the input. Here is an example:
```

```python
import litserve as ls
from litserve.specs.openai import ChatMessage

class OpenAISpecLitAPI(ls.LitAPI):
    def setup(self, device):
        self.model = None

    def predict(self, x):
        if isinstance(x[1]["content"], list):
            # handle mixed text and image input
            image_url = x[1]["content"][1]["image_url"]

            yield {"role": "assistant", "content": "\n\nThis image shows a wooden boardwalk extending through a lush green marshland."}
        else:
            yield {"role": "assistant", "content": "This is a generated output"}

    def encode_response(self, output: dict) -> ChatMessage:
        yield ChatMessage(role="assistant", content="This is a custom encoded output")

if __name__ == "__main__":
    server = ls.LitServer(OpenAISpecLitAPI(), spec=ls.OpenAISpec())
    server.run(port=8000)
```

> In this case, `predict` is expected to take an input with the following shape:

Reviewer: ```suggestion
When using `OpenAISpec`, `predict` is expected to take an input with the following shape:
```

- Text Input Example:
  ```
  [{"role": "system", "content": "You are a helpful assistant."},
   {"role": "user", "content": "Hello there"},
   {"role": "assistant", "content": "Hello, how can I help?"},
   {"role": "user", "content": "What is the capital of Australia?"}]
  ```
- Mixed Text and Image Input Example:

  ```
  [{"role": "system", "content": "You are a helpful assistant."},
   {
   "role": "user",
   "content": [
                  {"type": "text", "text": "What's in this image?"},
                  {
                      "type": "image_url",
                      "image_url": "https://upload.wikimedia.org/wikipedia/commons/thumb/d/dd/Gfp-wisconsin-madison-the-nature-boardwalk.jpg/2560px-Gfp-wisconsin-madison-the-nature-boardwalk.jpg",
                  },
              ]

  },
   {"role": "assistant", "content": "This image shows a wooden boardwalk extending through a lush green marshland."},

>    {"role": "user", "content": "How is the weather depicted in the image?"}]

Reviewer: ```suggestion
   {"role": "user", "content": "What is the weather like in the image?"}]
```
